rclcpp/test/rclcpp/test_time_source.cpp

```diff
@@ -695,7 +694,7 @@ class ClockThreadTestingNode : public rclcpp::Node
     this->set_parameter(rclcpp::Parameter("use_sim_time", true));
 
     // Create a 100ms timer
-    timer_ = this->create_wall_timer(
+    timer_ = this->create_timer(
```

Comment:
We should consider reverting this, even though the reversion should be a no-op. I get that this timer sets up the periodic publishing of `/clock` to drive peer nodes. For this reason a wall timer makes more sense. However, my understanding is that since the parameter`use_sim_time` is not set on this node, it should fall back to a wall timer. Thus, both `create_timer` and `create_wall_timer` are equivalent here.

Reply:
If I understand the tests correctly, we want this to be a wall timer so I would just revert it to be safe.

Reply:
yes, using `create_wall_timer()` here makes more sense I think

Reply:
Can you double check this is what you want!

> We should consider reverting this, even though the reversion should be a no-op. I get that this timer sets up the periodic publishing of /clock to drive peer nodes' perception of time (which we call "ROS time"). For this reason a wall timer makes more sense. However, my understanding is that since the parameter use_sim_time is not set on this node, create_timer should simply fall back to a wall timer. Thus, both create_timer and create_wall_timer are equivalent here.

The above message I sent was wrong, as I incorrectly thought this change was to  the `SimClockPublisherNode` code, which it is not. If you look at how this code is used in the unit test , we have a `SimClockPublisherNode` that publishes time on `/clock` and a `ClockThreadTestingNode` that consumes this time. I'm fairly sure that we want a `create_timer` here to prove that a `/clock` driven sense of time has a callback within which you can make subsequent calls to `this->now()`. If we really wanted to test a wall timer, there would be no need to spin up a `SimClockPublisherNode` instance.

Reply:
Yes, that seems to be the case.
I'm not sure what was the test doing before as `create_wall_timer()` never used sim time.

Reply:
[The comments](https://github.com/ros2/rclcpp/blob/b953bdddf8de213b4a051ecf2d668dad65ff9f89/rclcpp/test/rclcpp/test_time_source.cpp#L739-L742) also say that this test should be running with `use_clock_thread=true`, and that's not the case.

When I modify the test to do what's supposed to do, it's super flaky.
IMO, it's fair to comment out the test and create a new issue for it.
@jacobperron any opinion?

Reply:
I commented out the test as I think the problem is the test itself, see https://github.com/ros2/rclcpp/pull/1975/commits/009312185f959d93d0dde78b4a53d294aec68daa

Reply:
I thought the test was checking that the wall timer and simulation time updates were not interfering with each other. But, I'm not sure. I'm okay with disabling the flaky test. Maybe we can create a GitHub ticket to look into it (maybe it is just not a well-defined test).

Reply:
Based on the comments, the test name, and the code; it seems that the test is supposed to be checking that the clock is updated when `use_clock_thread=True` even if you're blocked in a callback.
The part that's missing is that `use_clock_thread` was never actually enabled.
Enabling that doesn't solve the issue, as it seems something is not working well.

@anaelle-sw I think you added this feature, could you take a look at the issue?